### PR TITLE
Added functional mini library

### PR DESCRIPTION
**Pull request by OlegAdamovich**

### CODE REVIEW

File: arrayMethodsLibrary.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const numbers = [100, 0, 250, 1000, 500];
+
+const arrayMethodsLibrary = {
+   first(array) {
+      return array[0];
```
Comment (mhryshkin) [inline arrayMethodsLibrary.js:5]: what if array `null`  or `[]`?

File: arrayMethodsLibrary.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+   },
+   last(array) {
```
Comment (mhryshkin) [inline arrayMethodsLibrary.js:7]: what if array null or []?

File: arrayMethodsLibrary.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      return array[array.length - 1];
+   },
+   skip(array, number) {
+      array.splice(0, number);
```
Comment (mhryshkin) [inline arrayMethodsLibrary.js:11]: no splice

File: arrayMethodsLibrary.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      return array;
+   },
+   take(array, number) {
+      return array.slice(0, number)
```
Comment (mhryshkin) [inline arrayMethodsLibrary.js:15]: no slice

File: typesLibrary.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+const typesLibrary = {
+    isArray(arg) {
+        return Array.isArray(arg)
```
Comment (mhryshkin) [inline typesLibrary.js:3]: no `Array.isArray`

File: typesLibrary.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    isBoolean(arg) {
+        return typeof arg === 'boolean'
+    },
+    isDate(arg) {
+        return typeof arg === 'object'
```
Comment (mhryshkin) [inline typesLibrary.js:9]: what is it??

File: typesLibrary.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        return typeof arg === 'string'
+    },
+    isFunction(arg) {
+        return typeof arg === 'function'
+    },
+    isUndefined(arg) {
+        return typeof arg === 'undefined'
+    },
+    isNull(arg) {
+        return typeof arg === 'object'
```
Comment (mhryshkin) [inline typesLibrary.js:24]: are you kidding me?

File: arrayMethodsLibrary.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+   first() {
+      this.value = this.value[0];
+      return this
+   },
+   last() {
+      this.value = this.value[this.value.length - 1];
+      return this
+   },
+   skip(number) {
+      this.value.splice(0, number);
```
Comment (mhryshkin) [inline arrayMethodsLibrary.js:30]: no splice

File: arrayMethodsLibrary.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+      return this
+   },
+   take(number) {
```
Comment (mhryshkin) [inline arrayMethodsLibrary.js:33]: no slice

File: typesLibrary.js (hunk 5 above)
Comment (OlegAdamovich) [reply typesLibrary.js:3]: return arg instanceof Array

File: typesLibrary.js (hunk 7 above)
Comment (OlegAdamovich) [reply typesLibrary.js:24]: lol
return arg === null

File: typesLibrary.js (hunk 6 above)
Comment (OlegAdamovich) [reply typesLibrary.js:9]: arg instanceof Date

File: arrayMethodsLibrary.js (hunk 1 above)
Comment (OlegAdamovich) [reply arrayMethodsLibrary.js:5]: if (array && array.length) {
        logic...
}

File: arrayMethodsLibrary.js (hunk 2 above)
Comment (OlegAdamovich) [reply arrayMethodsLibrary.js:7]: if (array && array.length) {
    logic...
}

File: arrayMethodsLibrary.js (hunk 3 above)
Comment (OlegAdamovich) [reply arrayMethodsLibrary.js:11]: Redone to for loop

File: arrayMethodsLibrary.js (hunk 4 above)
Comment (OlegAdamovich) [reply arrayMethodsLibrary.js:15]: Redone to for loop

File: arrayMethodsLibrary.js (hunk 8 above)
Comment (OlegAdamovich) [reply arrayMethodsLibrary.js:30]: fixed

File: arrayMethodsLibrary.js (hunk 9 above)
Comment (OlegAdamovich) [reply arrayMethodsLibrary.js]: fixed

**Review by MakaryHryshkin — APPROVED** [review]

**Pull request merged**